### PR TITLE
Ensure proper request id usage and logging

### PR DESCRIPTION
**Pull request by DRvanR**

### CODE REVIEW

**Comment by DRvanR** [comment]
@arothuis

**Comment by DRvanR** [comment]
@arothuis

File: src/Request/RequestId.php [hunk 1]
```diff
@@ -57,9 +57,19 @@ public function get()
         return $this->requestId;
     }
 
-    public function set($requestId)
+    /**
+     * We allow overwriting the RequestId so that we can inject a RequestId from a header when log statements already
+     * have been made - which would cause an exception otherwise. The use-case here is the Stepup-Middleware
+     * application, this application receives API-calls, but by then a logs messages have already been written.
```
Comment (arothuis) [inline src/Request/RequestId.php]: Nit: "a log message" or "log messages" or "a log's messages"

File: src/Request/RequestId.php [hunk 2]
```diff
@@ -57,9 +57,19 @@ public function get()
...
+     * However, for the sake of correlation we do want to use the log to show the correct request_id when actually
+     * handling the request.
```
Comment (arothuis) [inline src/Request/RequestId.php]: :+1: for the comment

**Comment by arothuis** [comment]
👍

**Pull request merged**